### ISSUE TITLE
Higher dimension support for `MultiLableSoftMarginLoss`

### ISSUE THREAD

**Issue by zegzag** [pytorch/pytorch#26228]
## 🚀 Feature
In the current version, input and target shape of `MultiLableSoftMarginLoss` is (N, C), which only support  one dimension tensor for batch data. I wonder if this can support high dimension as shape of input and target can be (N, C, d1, d2, ...),  just like the `CrossEntropyLoss`.

## Motivation
Motivation comes from the sequence tagging task in NLP.   In traditional cases, if we get padding length for `P`,  batch size for `N`, classification amount for `C`. We can use `CrossEntropyLoss` for sequence tagging loss as input and target shape is (N, C, P) for batch training. However, this only works when each word belongs to ONE of the classes. If I want to train the data with multi labels for each word, I must use Multi-Label loss function such as `MultiLableSoftMarginLoss`. And in this case, I have to calculate the loss one sentence at a time as input and target shape is (P, C). Batch calculation is disabled in this case, which makes the process time-consuming.
